orttraining/orttraining/python/training/api/module.py

```diff
@@ -80,11 +80,11 @@ def eval(self):
         """
         return self.train(False)
 
-    def reset_grad(self):
+    def lazy_reset_grad(self):
         """
-        Resets the gradient of the parameters.
+        Lazily resets the gradient of the parameters.
```

Comment:
can you add a little more explanation for what does lazy mean here... similar to C API comments.